File: EEFRT Demo iOS/EEFRT Demo/Utills/GameConfigUtils.swift
import Foundation
import SwiftyUserDefaults

struct GameConfigUtils {
    private static let rewardPaymentThreshold = 0.8

    static func rewardThresholdReached() -> Bool {
        guard let cache = Defaults.gameCache,
              let nTrials = cache.randTrialsIdx?.count else { return false }

        let nTrialsToReachThreshold = Int(Double(nTrials) * rewardPaymentThreshold)
        return cache.trialNumber + 1 >= nTrialsToReachThreshold
    }
}

File: EEFRT Demo Android/app/src/main/java/ai/a2i2/conductor/effrtdemoandroid/util/GameConfigUtils.kt
package ai.a2i2.conductor.effrtdemoandroid.util

import ai.a2i2.conductor.effrtdemoandroid.persistence.GameStorage
import android.content.Context

class GameConfigUtils {
    companion object {
        const val REWARD_PAYMENT_THRESHOLD = 0.8

        fun rewardThresholdReached(context: Context): Boolean {
            val trialNumber = GameStorage(context).cachedGameState?.trialNumber?.plus(1) ?: 0
            val nTrials = GameStorage(context).cachedGameState?.randTrialsIdx?.size ?: 0
            val minTrialsCompleted = (nTrials * REWARD_PAYMENT_THRESHOLD).toInt()
            return trialNumber >= minTrialsCompleted
        }
    }
}
